src/main/kotlin/com/exactpro/th2/codec/fixng/FixNgCodec.kt

```diff
@@ -124,57 +150,169 @@ class FixNgCodec(messages: List<Message>, settings: FixNgCodecSettings) : IPipel
         return MessageGroup(messages)
     }
 
-    private fun Field.decode(source: ByteBuf, target: MutableMap<String, Any>, value: String, tag: Int) {
+    private fun Field.decode(source: ByteBuf, target: MutableMap<String, Any>, value: String, tag: Int, context: IReportingContext) {
         val previous = when (this) {
-            is Primitive -> target.put(name, value)
-            is Group -> target.put(name, decode(source, value.toIntOrNull() ?: error("Invalid $name group counter ($tag) value: $value")))
+            is Primitive -> {
+                val decodedValue = when (primitiveType) {
```

Comment:
issue(blocking): this is a dirty codec. We can send a message that contains a string in a tag that should be an integer, for example. This will result in a decoding error as far as I understand (please correct me if I am wrong here). We probably should check whether the message contains `ENCODE_MODE_PROPERTY_NAME` property and do not perform value validation

Reply:
According to task TH2-5219, it is required to implement the dirty mode only for encoding. Also, the name of the property indicates that it relates only to encoding.

Reply:
Okay, but let's think for a second: how does the user get the messages he or she sent? Right now it is impossible to decode the message like this and the user will always get an error in response from codec

Reply:
Yes, it is impossible to decode incorrect messages, and it should not be possible. In my opinion, that's how it should work: in 'general mode', incorrect messages are not processed and an error is generated. In 'dirty mode', incorrect messages are processed and a warning is generated.

Reply:
> In 'dirty mode', incorrect messages are processed and a warning is generated.

Yes, I agree. So, the codec will not report an error in case a "dirty" message is decoded (just a warning) and it will continue to parse the message and produce the decoded result. Right, @lumber1000 ?

Reply:
I think we can check if the message contains `ENCODE_MODE_PROPERTY_NAME` property and do not report errors for such messages - it is expected that the normal decoding should fail for them. All other messages should be decoded with all required validations